core/src/test/java/org/elasticsearch/bwcompat/OldIndexBackwardsCompatibilityIT.java

```diff
@@ -1,201 +0,0 @@
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-
-package org.elasticsearch.bwcompat;
-
-import org.apache.lucene.util.LuceneTestCase;
-import org.apache.lucene.util.TestUtil;
-import org.elasticsearch.Version;
-import org.elasticsearch.VersionTests;
-import org.elasticsearch.cluster.metadata.IndexMetaData;
-import org.elasticsearch.cluster.metadata.MetaData;
-import org.elasticsearch.common.io.FileSystemUtils;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.xcontent.XContentBuilder;
-import org.elasticsearch.common.xcontent.XContentParser;
-import org.elasticsearch.common.xcontent.XContentType;
-import org.elasticsearch.gateway.MetaDataStateFormat;
-import org.elasticsearch.plugins.Plugin;
-import org.elasticsearch.test.ESIntegTestCase;
-import org.elasticsearch.test.InternalSettingsPlugin;
-import org.elasticsearch.test.OldIndexUtils;
-import org.elasticsearch.test.VersionUtils;
-import org.junit.AfterClass;
-import org.junit.Before;
-
-import java.io.IOException;
-import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.Locale;
-import java.util.SortedSet;
-import java.util.TreeSet;
-
-import static org.elasticsearch.test.OldIndexUtils.getIndexDir;
-
-// needs at least 2 nodes since it bumps replicas to 1
-@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
-@LuceneTestCase.SuppressFileSystems("ExtrasFS")
-public class OldIndexBackwardsCompatibilityIT extends ESIntegTestCase {
-    // TODO: test for proper exception on unsupported indexes (maybe via separate test?)
-    // We have a 0.20.6.zip etc for this.
-
-
-    @Override
-    protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Arrays.asList(InternalSettingsPlugin.class);
-    }
-
-    List<String> indexes;
-    List<String> unsupportedIndexes;
-    static String singleDataPathNodeName;
-    static String multiDataPathNodeName;
-    static Path singleDataPath;
-    static Path[] multiDataPath;
-
-    @Before
-    public void initIndexesList() throws Exception {
-        indexes = OldIndexUtils.loadDataFilesList("index", getBwcIndicesPath());
-        unsupportedIndexes = OldIndexUtils.loadDataFilesList("unsupported", getBwcIndicesPath());
-    }
-
-    @AfterClass
-    public static void tearDownStatics() {
-        singleDataPathNodeName = null;
-        multiDataPathNodeName = null;
-        singleDataPath = null;
-        multiDataPath = null;
-    }
-
-    @Override
-    public Settings nodeSettings(int ord) {
-        return OldIndexUtils.getSettings();
-    }
-
-    public void testAllVersionsTested() throws Exception {
-        SortedSet<String> expectedVersions = new TreeSet<>();
```

Comment:
For posterity: this is handled by the gradle logic that scrapes the version file and a test in `VersionUtilsTests` that verifies that the gradle logic and VersionUtils line up.